great_tables/_spanners.py

```diff
@@ -706,6 +706,8 @@ def cols_width(self: GTSelf, cases: dict[str, str] | None = None, **kwargs: str)
     _assert_list_is_subset(mod_columns, set_list=column_names)
 
     for col, width in new_cases.items():
+        if not isinstance(width, str):
+            raise ValueError(f"Width must be a string. Column {col} width received a {type(width)}")
```

Comment:
@rich-iannone I noticed the annotations require width to be a string, but it's not enforced. WDYT of raising an error if a non-string is passed? If it's okay for users to pass a number, we could always coerce it to a string here instead!

Reply:
I definitely prefer that a string be the required input here. This is a good improvement, thanks!

Reply:
Upon discussion with @machow , we decided that a warning is more suitable here.

Reply:
My understanding is the R program allows integers, so we'll fire a deprecation warning and coerce to a string for a while, just in case this path gets used by folks....!